### PR TITLE
Update JSON update format handling for parity with RSolr 1.x XML handling

### PR DESCRIPTION
**Pull request by cbeer**

### CODE REVIEW

**Comment by cbeer** [comment]
This provides full compatibility with the sunspot test suite.. it deserves some tests, but I'm digging myself out of an unrelated gem problem right now.

File: lib/rsolr.rb [hunk 1]
```diff
@@ -1,7 +1,8 @@
 module RSolr
   
   Dir.glob(File.expand_path("../rsolr/*.rb", __FILE__)).each{|rb_file| require(rb_file)}
-  
+  module Char; end
```
Comment (jcoyne) [inline lib/rsolr.rb]: Add a `self.included` method that generates a deprecation warning?

Comment (cbeer) [inline lib/rsolr.rb]: 👌  good idea.

Comment (cbeer) [inline lib/rsolr.rb]: Fixed.

**Pull request merged**